### PR TITLE
Add docs for `JoinPosition` for concentrated liquidity

### PR DESCRIPTION
**Pull request by mattverse**
This PR adds docs for the `JoinPosition` API and the math and equations on how we solve the amount of token0 and token1 given the amount to join and the upper tick and the lower tick

### CODE REVIEW

File: docs/architecture/concentrated-liquidity-adr.md [hunk 1]
```diff
@@ -142,18 +142,41 @@ TODO: figure out routing logic:
 > As an LP, I want to provide liquidity in ranges so that I can achieve greater capital efficiency
 
 This a basic function that should allow LPs to provide liquidity in specific ranges
-to a pool. We will disucss each of these functions in more detail in the following sections.
+to a pool. 
 
-##### Msg
+The liquidity of the pool is consisted of both asset0 and asset1. The bucket that includes the current price is what we so call the "liquidity of the pool". Any other liquidity below or above the current price is consisted only of a single asset.
```
Comment (czarcas7ic) [inline docs/architecture/concentrated-liquidity-adr.md:147]:
```suggestion
A pool's liquidity is consisted of two assets: asset0 and asset1. In all pools, asset0 will be the lexicographically smaller of the two assets. At the current price tick, the bucket at this tick consists of a mix of both asset0 and asset1 and is called the virtual liquidity of the pool (or "L" for short). Any positions set below the current price are consisted solely of asset0 while positions above the current price only contain asset1.
```

File: docs/architecture/concentrated-liquidity-adr.md [hunk 2]
```diff
@@ -142,18 +142,41 @@ TODO: figure out routing logic:
...
+
+Therefore in `Mint`, we can either provide liquidity above or below the current price, which would act as limit orders, or decide to provide liquidity at current price. 
+
+As declared in the API for mint, users are to provide the upper tick and the lower tick they want to provide the liquidity in. The users are also prompted to provide the amount of token0 and token1 they desire to receive. The liquidity that needs to be provided for the token0 and token1 amount provided would be then calculated by the following methods.
```
Comment (czarcas7ic) [inline docs/architecture/concentrated-liquidity-adr.md:151]:
```suggestion
As declared in the API for mint, users provide the upper and lower tick to denote the range they want to provide the liquidity for. The users are also prompted to provide the amount of token0 and token1 they desire to receive. The liquidity that needs to be provided for the token0 and token1 amount provided would be then calculated by the following methods: 
```

**Review by czarcas7ic — APPROVED** [review]

File: docs/architecture/concentrated-liquidity-adr.md [hunk 3]
```diff
@@ -142,18 +142,41 @@ TODO: figure out routing logic:
 > As an LP, I want to provide liquidity in ranges so that I can achieve greater capital efficiency
 
 This a basic function that should allow LPs to provide liquidity in specific ranges
-to a pool. We will disucss each of these functions in more detail in the following sections.
+to a pool. 
 
-##### Msg
+A pool's liquidity is consisted of two assets: asset0 and asset1. In all pools, asset0 will be the lexicographically smaller of the two assets. At the current price tick, the bucket at this tick consists of a mix of both asset0 and asset1 and is called the virtual liquidity of the pool (or "L" for short). Any positions set below the current price are consisted solely of asset0 while positions above the current price only contain asset1.
+
+Therefore in `Mint`, we can either provide liquidity above or below the current price, which would act as limit orders, or decide to provide liquidity at current price.
```
Comment (p0mvn) [inline docs/architecture/concentrated-liquidity-adr.md:149]:
```suggestion
Therefore in `Mint`, we can either provide liquidity above or below the current price, which would act as range (limit) orders or decide to provide liquidity at the current price. 
```

File: docs/architecture/concentrated-liquidity-adr.md [hunk 4]
```diff
@@ -142,18 +142,41 @@ TODO: figure out routing logic:
...
+
+As declared in the API for mint, users provide the upper and lower tick to denote the range they want to provide the liquidity for. The users are also prompted to provide the amount of token0 and token1 they desire to receive. The liquidity that needs to be provided for the token0 and token1 amount provided would be then calculated by the following methods: 
+
+Liquidity needed for token0:
+$$L = \frac{\Delta x \sqrt{P_u} \sqrt{P_l}}{\sqrt{P_u} - \sqrt{P_l}}$$
+
+Liquidity needed for token1:
+$$L = \frac{\Delta y}{\sqrt{P_u}-\sqrt{P_l}}$$
+
+With the larger liquidity including the smaller liquidity, we take the smaller liquidity calculated for both token0 and token1 and use that as the liquidity throughout the rest of the joining process. Note that the liquidity used here does not represent an amount of a specific token, but the liquidity of the pool itself, represented in sdk.Int.
```
Comment (p0mvn) [inline docs/architecture/concentrated-liquidity-adr.md:159]: Can we elaborate on "larger liquidity including smaller" liquidity?

**Review by p0mvn — APPROVED** [review]

**Pull request merged**